bp_common/software/py/nbf.py

```diff
@@ -217,8 +246,10 @@ def dump(self):
   parser.add_argument("--checkpoint", dest='checkpoint_file', metavar='sample.nbf',help='checkpoint nbf file')
   parser.add_argument('--skip_zeros', dest='skip_zeros', action='store_true', help='skip zero DRAM entries')
   parser.add_argument('--addr_width', type=int, default=40, help='Physical address width')
+  parser.add_argument('--data_width', type=int, default=64, help='Data width')
 
   args = parser.parse_args()
 
-  converter = NBF(args.ncpus, args.ucode_file, args.mem_file, args.checkpoint_file, args.config, args.skip_zeros, args.addr_width)
+  converter = NBF(args.ncpus, args.ucode_file, args.mem_file, args.checkpoint_file, args.config,
+          args.skip_zeros, args.addr_width, args.data_width)
```

Comment:
Not necessarily part of this change, but for configurations that don't use the ucode CCE, does the generated NBF still contain commands to load the microcode? It looks like this script would skip that step if ucode_file is not provided, but IIRC the rest of the build flow may always generate the ucode file for simulation, so it may always be included.

Reply:
Yep, you're right: https://github.com/black-parrot/black-parrot/issues/840